### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by e-edling-ericsson**
### Description of the Change

If you want to try it out, please check the travis file.
- Adding the rest of the integrationtests. They should now all work as soon as EI is able to use jenkins with crumbs which is in the PR #220. Adding this PR until then in order to fix code comments.
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: 
Erik Edling erik.edling@ericsson.com

### CODE REVIEW

File: src/integrationtests/java/util/IntegrationTestBase.java [hunk 1]
```diff
@@ -44,42 +46,61 @@
...
+    private static final String EIFFEL_INTELLIGENCE_DATABASE_NAME = "eiffel_intelligence";
 
     @Autowired
-    private MongoDBHandler mongoDBHandler;
+    public ObjectHandler objectHandler;
 
     @Value("${spring.data.mongodb.database}")
     private String database;
-
     @Value("${event_object_map.collection.name}")
```
Comment (evasiba-ericsson) [inline src/integrationtests/java/util/IntegrationTestBase.java]: this is also fetched with System.getProperty. Use only one

File: src/integrationtests/java/util/IntegrationTestBase.java [hunk 2]
```diff
@@ -192,4 +236,18 @@ private void checkResult(final Map<String, JsonNode> checkData) throws IOExcepti
...
+        CachingConnectionFactory cf = new CachingConnectionFactory();
+        cf.setHost(rabbitMqHost);
+        cf.setPort(rabbitMqPort);
+        cf.setUsername(rabbitMqUsername);
+        cf.setPassword(rabbitMqPassword);
+
+        RabbitTemplate rabbitTemplate = new RabbitTemplate(cf);
+        rabbitTemplate.setExchange(exchangeName);
+        rabbitTemplate.setRoutingKey(bindingKey);
+        rabbitTemplate.setQueue(consumerName);
```
Comment (evasiba-ericsson) [inline src/integrationtests/java/util/IntegrationTestBase.java]: We agreed that you should not send to a specific queue. but only to an exchange. Please remove set queue.

Comment (e-edling-ericsson) [reply src/integrationtests/java/util/IntegrationTestBase.java]: I think i missunderstood, will remove :)

File: src/integrationtests/java/util/IntegrationTestBase.java (hunk 1 above)
Comment (e-edling-ericsson) [reply src/integrationtests/java/util/IntegrationTestBase.java]: They get null if i use System.getProperty. I can make the other ones using System.getProperty to use "@Value" instead. I guess the spring properties aren't set as system property but system properties seems to be set to spring properties

File: pom.xml [hunk 3]
```diff
@@ -450,35 +451,38 @@
...
-                                <exclude>**/functionaltests/*</exclude>
-                            </excludes>
-                            <additionalClasspathElements>
-                                <additionalClasspathElement>${basedir}/target/classes</additionalClasspathElement>
-                            </additionalClasspathElements>
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
+                  <groupId>org.apache.maven.plugins</groupId>
```
Comment (a-breid-ericsson) [inline pom.xml]: 5 spaces not 4

File: pom.xml [hunk 4]
```diff
@@ -450,35 +451,38 @@
...
+                          <id>integration-test</id>
+                          <goals>
+                              <goal>integration-test</goal>
+                              <goal>verify</goal>
+                          </goals>
+                          <phase>integration-test</phase>                          
+                      </execution>
+                  </executions>
+                  <configuration>
+                              <forkCount>3</forkCount>
```
Comment (a-breid-ericsson) [inline pom.xml]: Many spaces too many

File: src/integrationtests/java/com/ericsson/ei/integrationtests/FlowStepsIT.java [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    @Given("^the resulting aggregated object \"([^\"]*)\";$")
+    public void the_resulting_aggregated_object(String aggregatedObjectFilePath) throws Throwable {
+        this.aggregatedObjectFilePath = aggregatedObjectFilePath;
+    }
+
+    @Given("^the expected aggregated object ID is \"([^\"]*)\"$")
+    public void the_expected_aggregated_object_ID_is(String aggregatedObjectID) throws Throwable {
+        // Write code here that turns the phrase above into concrete actions
```
[the remark this answers is not in the document]
Comment (e-edling-ericsson) [reply src/integrationtests/java/com/ericsson/ei/integrationtests/FlowStepsIT.java]: Done

File: src/integrationtests/java/util/IntegrationTestBase.java [hunk 6]
```diff
@@ -178,18 +226,52 @@ protected void waitForEventsToBeProcessed(int eventsCount) throws InterruptedExc
         }
     }
 
-    private void checkResult(final Map<String, JsonNode> checkData) throws IOException {
+    // count documents that were processed
```
[the remark this answers is not in the document]
Comment (e-edling-ericsson) [reply src/integrationtests/java/util/IntegrationTestBase.java]: Done

File: src/integrationtests/java/util/IntegrationTestBase.java [hunk 7]
```diff
@@ -178,18 +226,52 @@ protected void waitForEventsToBeProcessed(int eventsCount) throws InterruptedExc
...
+    private long countProcessedEvents(String database, String collection) {
+        MongoClient mongoClient = null;
+        mongoClient = mongoDBHandler.getMongoClient();
+        MongoDatabase db = mongoClient.getDatabase(database);
+        MongoCollection table = db.getCollection(collection);
+        return table.count();
+    }
+
+    private void checkResult(final Map<String, JsonNode> checkData) throws IOException, URISyntaxException {
```
[the remark this answers is not in the document]
Comment (e-edling-ericsson) [reply src/integrationtests/java/util/IntegrationTestBase.java]: Done

File: src/integrationtests/java/util/IntegrationTestBase.java [hunk 8]
```diff
@@ -178,18 +226,52 @@ protected void waitForEventsToBeProcessed(int eventsCount) throws InterruptedExc
...
+            ResponseEntity<String> response = httpRequest.performRequest();
+            JsonNode body = objectMapper.readTree(response.getBody());
+            //The response contains the aggregated object as a jsonstring. Makes it this wierd to get out.
+            JsonNode responseEntity = objectMapper.readTree(body.get("responseEntity").asText());
+            JsonNode actualJSON = responseEntity.get(0);
             JSONAssert.assertEquals(expectedJSON.toString(), actualJSON.toString(), false);
         }
     }
+
+    private RabbitTemplate createRabbitMqTemplate() {
```
[the remark this answers is not in the document]
Comment (e-edling-ericsson) [reply src/integrationtests/java/util/IntegrationTestBase.java]: Very self explaining method so i'll skip it for this one

File: pom.xml (hunk 3 above)
Comment (e-edling-ericsson) [reply pom.xml]: done

File: pom.xml (hunk 4 above)
Comment (e-edling-ericsson) [reply pom.xml]: done

**Pull request merged**